<?xml version="1.0" encoding="utf-8"?>
<!--
 Copyright (c) 2021  Carnegine Mellon University and IBM Corporation, and others

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
-->

<launch>
  <arg name="map_frame" default=""/>
  <arg name="camera_link_frame" default=""/>
  <arg name="camera_info_topic" default=""/>
  <arg name="image_rect_topic" default=""/>
  <arg name="depth_registered_topic" default=""/>
  <arg name="namespace" default="camera"/>
  
  <node namespace="$(var namespace)"
	name="detect_darknet_people_py" pkg="track_people_py" exec="detect_darknet_opencv_people.py" output="log">
    <param from="$(find-pkg-share track_people_py)/params/realsense.yaml" />

    <!--
    <param name="map_frame" value="$(var map_frame)"/>
    <param name="camera_id" value="$(var namespace)"/>
    <param name="camera_link_frame" value="$(var camera_link_frame)"/>
    <param name="camera_info_topic" value="$(var camera_info_topic)"/>
    <param name="image_rect_topic" value="$(var image_rect_topic)"/>
    <param name="depth_registered_topic" value="$(var depth_registered_topic)"/>
-->
    
    <param name="detection_threshold" value="0.25" />
    <param name="minimum_detection_size_threshold" value="50" />
    
    <param name="detect_config_file" value="$(find-pkg-share track_people_py)/scripts/yolov4.cfg" />
    <param name="detect_weight_file" value="$(find-pkg-share track_people_py)/scripts/yolov4.weights" />
    <param name="detect_label_file" value="$(find-pkg-share track_people_py)/scripts/coco.names" />
    <!-- cannot load params by include
	 <include file="$(find-pkg-share track_people_py)/launch/detect_common_darknet.launch.xml" />
    -->
  </node>
</launch>
